src/Assignment/Assignment.php

```diff
@@ -38,4 +48,86 @@ public function canonicalize(): string
         }
         return $canonical;
     }
+
+    /**
+     * Convert an Assignment to an Amplitude event
+     */
+    public function toEvent(): Event
+    {
+        $event = new Event('[Experiment] Assignment');
+        $event->userId = $this->user->userId;
+        $event->deviceId = $this->user->deviceId;
+        $event->eventProperties = [];
+        $event->userProperties = [];
+
+        $set = [];
+        $unset = [];
+        foreach ($this->variants as $flagKey => $variant) {
+            if (!$variant->key) {
+                continue;
+            }
+            $event->eventProperties["{$flagKey}.variant"] = $variant->key;
+            $version = $variant->metadata['flagVersion'] ?? null;
+            $segmentName = $variant->metadata['segmentName'] ?? null;
+            if ($version && $segmentName) {
+                $event->eventProperties["{$flagKey}.details"] = "v{$version} rule:{$segmentName}";
+            }
+            $flagType = $variant->metadata['flagType'] ?? null;
+            $default = $variant->metadata['default'] ?? false;
+            if ($flagType == FLAG_TYPE_MUTUAL_EXCLUSION_GROUP) {
```

Comment:
FLAG_TYPE_MUTUAL_EXCLUSION_GROUP is not defined so it throws a PHP error. I guess this if is coming from AssignmentService because this const is defined there but is not used.

Reply:
Many thanks for the comments! This has been resolved in the latest commit.